src/main/kotlin/com/github/imflog/schema/registry/Subject.kt

```diff
@@ -25,7 +33,27 @@ data class Subject(
         return this
     }
 
+    fun setMetadata(path: String): Subject {
+        val metadataContent = File(path).readText(Charsets.UTF_8)
+        metadata =  Gson().fromJson(metadataContent, Metadata::class.java)
+        return this
+    }
+
+    fun setRuleSet(path: String): Subject {
+        val ruleSetContent = File(path).readText(Charsets.UTF_8)
+        ruleSet = Gson().fromJson(ruleSetContent, RuleSet::class.java)
+        return this
+    }
+
+    fun setNormalized(normalize: Boolean): Subject {
+        this.normalize = normalize
+        return this
+    }
+
     // Used for data class destructuring
     operator fun component4() = references
     operator fun component5() = localReferences
-}
\ No newline at end of file
+    operator fun component6() = metadata
+    operator fun component7() = ruleSet
+    operator fun component8() = normalize
```

Comment:
As explained later, I think we should remove the de-structuring operator and go back to using a variable.